### PR TITLE
IDE0057 simplify substring

### PR DESCRIPTION
**Pull request by gerhardol**
## Proposed changes

VS suggested change, simplifies displayed code using range operator.
(Should be no change to runtime execution.)

https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0057

Typical like

```
diff --git a/BugReporter/GitHubUrlBuilder.cs b/BugReporter/GitHubUrlBuilder.cs
index 8377b..5b8b8 100644
--- a/BugReporter/GitHubUrlBuilder.cs
+++ b/BugReporter/GitHubUrlBuilder.cs
@@ -46,7 +46,7 @@ public GitHubUrlBuilder(IErrorReportUrlBuilder errorReportUrlBuilder)
             if (subject.Length > 69)
             {
                 // if the subject is longer than 70 characters, trim it
-                subject = subject.Substring(0, 66) + "...";
+                subject = subject[..66] + "...";
             }
 
             string urlEncodedError = _errorReportUrlBuilder.Build(exception, exceptionInfo, environmentInfo, additionalInfo);
```

## Test methodology 

Refactoring

## Merge strategy

I agree that the maintainer squash merge this PR (if the commit message is clear).

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

**Review by RussKie — APPROVED** [review]
I'm not used to this syntax, but I'm not going to stand in the way of progress :)

File: Externals/NetSpell.SpellChecker/Spelling.cs [hunk 1]
```diff
@@ -747,8 +747,8 @@ public void ReplaceWord()
             // if first letter upper case, match case for replacement word
             if (char.IsUpper(_words[replacedIndex].ToString(), 0))
             {
-                _replacementWord = _replacementWord.Substring(0, 1).ToUpper(CultureInfo.CurrentUICulture)
-                    + _replacementWord.Substring(1);
+                _replacementWord = _replacementWord[..1].ToUpper(CultureInfo.CurrentUICulture)
```
[the remark this answers is not in the document]
Comment (gerhardol) [reply Externals/NetSpell.SpellChecker/Spelling.cs:750]: char is handled in a different way than a string with len=1
_replacementWord = $"{char.ToUpper(_replacementWord[0], CultureInfo.CurrentUICulture)}{_replacementWord[1..]}";

File: Externals/NetSpell.SpellChecker/Spelling.cs [hunk 2]
```diff
@@ -287,15 +287,15 @@ private void ReplaceChars(List<Word> tempSuggestion)
...
+                string replacement = replacementChars[i][(split + 1)..];
 
                 int pos = CurrentWord.IndexOf(key, StringComparison.InvariantCulture);
                 while (pos > -1)
                 {
-                    string tempWord = CurrentWord.Substring(0, pos);
+                    string tempWord = CurrentWord[..pos];
                     tempWord += replacement;
-                    tempWord += CurrentWord.Substring(pos + key.Length);
+                    tempWord += CurrentWord[(pos + key.Length)..];
```
[the remark this answers is not in the document]
Comment (gerhardol) [reply Externals/NetSpell.SpellChecker/Spelling.cs:298]: Another fix in an external file - still shows up in searches so changing

File: Plugins/BuildServerIntegration/TeamCityIntegration/TeamCityAdapter.cs [hunk 3]
```diff
@@ -569,7 +569,7 @@ private Task<XDocument> GetFilteredBuildsXmlResponseAsync(string buildTypeId, Ca
         private static DateTime DecodeJsonDateTime(string dateTimeString)
         {
             var dateTime = new DateTime(
-                    int.Parse(dateTimeString.Substring(0, 4)),
+                    int.Parse(dateTimeString[..4]),
                     int.Parse(dateTimeString.Substring(4, 2)),
```
[the remark this answers is not in the document]
Comment (gerhardol) [reply Plugins/BuildServerIntegration/TeamCityIntegration/TeamCityAdapter.cs]: Using .AsSpan(), no allocation at all needed
Still do not see why the autoaction did not detect these calls.

**Review by mstv — APPROVED** [review]

**Pull request merged**